### PR TITLE
fix loading data >2d without axes names + misc fixes 

### PR DESCRIPTION
**Pull request by gdementen**
After two days of pulling my hair, I finally got the full test suite to pass on python2.7 and python3.6+. At least 3 hours of which
were spent because I did not realize my version of pandas was different in my Python2.7 environment than on my other environments (doh!).

I have not tested python3.5 but let's hope it behaves nicely :)

Also I only tested Python3.8 as far as packages for our optional dependencies are available, ie all tests depending on either 
xlwings or pytables were not run because those packages are not available yet.

### CODE REVIEW

**Comment by alixdamman** [comment]
I can't wait for dropping Python 2.7

**Comment by gdementen** [comment]
After this... Me too :)

**Comment by gdementen** [comment]
I guess issue #830 is the only one worth a mention in the changelog. I will add this tomorrow.

File: larray/inout/excel.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 __all__ = ['read_excel']
 
 
+# CHECK: used "# doctest: +SKIP" for all tests because they work only if xlrd (an *optional* dependency) is installed
 # TODO: remove '# doctest: +SKIP' next to arr.info when Python 2.7 will be dropped
```
Comment (alixdamman) [inline larray/inout/excel.py]: remove the TODO

**Comment by gdementen** [comment]
@alixdamman are you ok with https://github.com/larray-project/larray/pull/837/commits/f558e41e7e18b8fcac5f1536aa2f84d191643067 ?

**Comment by gdementen** [comment]
I didn't rerun the test suite after rebasing. I should have of course :)

**Comment by alixdamman** [comment]
I'm ok with https://github.com/larray-project/larray/commit/f558e41e7e18b8fcac5f1536aa2f84d191643067

**Pull request merged**